generator/views/operational_monitoring_view.py

```diff
@@ -0,0 +1,102 @@
+"""Class to describe an Operational Monitoring View."""
+
+from textwrap import dedent
+from typing import Any, Dict, List, Optional, Set
+
+from .ping_view import PingView
+from .view import ViewDict
+
+# These are fields we don't need in our view
+EXCLUDED_FIELDS: Set[str] = {
+    "submission_date",
+    "client_id",
+    "build_id",
+    "branch",
+    "metrics",
+}
+
+
+class OperationalMonitoringView(PingView):
+    """A view on a operational monitoring table."""
+
+    type: str = "operational_monitoring_view"
+    dimensions: List[Dict[str, str]] = [
+        {
+            "name": "build_id",
+            "type": "date",
+            "sql": "PARSE_DATE('%Y%m%d', CAST(${TABLE}.build_id AS STRING))",
+        },
+        {"name": "branch", "type": "string", "sql": "${TABLE}.branch"},
+        {"name": "probe", "type": "string", "sql": "${TABLE}.name"},
+    ]
+
+    measures: List[Dict[str, str]] = [
+        {
+            "name": "median",
+            "type": "number",
+            "sql": dedent(
+                """
+            mozfun.hist.percentiles(
+              mozfun.hist.merge(
+                ARRAY_AGG(
+                  ${TABLE}.histograms IGNORE NULLS
+                )
+              ),
+              [0.5]
+            )[SAFE_OFFSET(0)].value
+            """
+            ),
+        }
+    ]
+
+    def __init__(self, namespace: str, name: str, tables: List[Dict[str, str]]):
+        """Create instance of a OperationalMonitoringView."""
+        super().__init__(namespace, name, tables)
+
+    @classmethod
+    def from_dict(klass, namespace: str, name: str, _dict: ViewDict) -> PingView:
+        """Get a OperationalMonitoringView from a dict representation."""
+        return OperationalMonitoringView(namespace, name, _dict["tables"])
+
+    def to_lookml(self, bq_client, v1_name: Optional[str]) -> Dict[str, Any]:
+        """Get this view as LookML."""
+        if len(self.tables) == 0:
+            raise Exception(
+                (f"Operational Monitoring view {self.name} " f"has no tables")
+            )
+
+        reference_table = self.tables[0]["table"]
+
+        schema = bq_client.get_table(reference_table).schema
+        for field in schema:
+            if field.name in EXCLUDED_FIELDS:
+                continue
+
+            self.dimensions.append(
+                {
+                    "name": field.name,
+                    "type": field.field_type,
+                    "sql": "${TABLE}." + f"{field.name}",
+                }
+            )
```

Comment:
This should either be using:
1. The utils in https://github.com/mozilla/lookml-generator/blob/main/generator/views/lookml_utils.py. See e.g. `ping_view` for how this works
2. Create a `table_view` for each one of these, and extend that view for these views. Then you only need to add the measures (and any derived dimensions). See e.g. `growth_accounting` for an example of this

Reply:
ok I used 1)